evm/src/main/java/org/hyperledger/besu/evm/precompile/AltBN128MulPrecompiledContract.java

```diff
@@ -40,6 +42,8 @@ public class AltBN128MulPrecompiledContract extends AbstractAltBnPrecompiledCont
 
   private static final Bytes POINT_AT_INFINITY = Bytes.repeat((byte) 0, 64);
   private final long gasCost;
+  private static final Cache<Integer, PrecompileInputResultTuple> bnMulCache =
+      Caffeine.newBuilder().maximumSize(1000).build();
```

Comment:
Not really urgent or mandatory but it would be nice to add metrics for it to have the cache hit ratio. Check : https://github.com/hyperledger/besu/blob/8cddcfd9550f37bd5a78888254bf7ac71bd3fdea/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/bonsai/cache/BonsaiCachedMerkleTrieLoader.java#L43

Reply:
Good idea.  I will add that

Reply:
Added hit/miss/false positive counters per precompile.  ECRECOVER has 0% false positive, but eip 196, 197, and blake2 seem to need a better hashcode implementation to limit the false positives.

Reply:
for blake2 and altbn128 methods, the false positive vs hit ratio is not good:
<img width="984" alt="image" src="https://github.com/user-attachments/assets/d4ee6056-8dec-4c1b-88c6-da8782c6a0bf" />

Reply:
ecrecover, which is hit much more frequently has zero false positives, and a good enough cache hit ratio to make it worthwhile:
<img width="1009" alt="image" src="https://github.com/user-attachments/assets/113a0fe8-feca-44b0-b76f-239536a726ec" />